Small features implemented

onuridrisoglu opened the pull request:
- Login user is moved to session
- Menubar filters
- report distribution bar

onuridrisoglu committed:
Report Distribution Bar Addon is integrated

onuridrisoglu committed:
rebel.xml is removed from sc

onuridrisoglu committed:
Report Distribution Bar Addon

onuridrisoglu committed:
login user is kept in the vaadin session

onuridrisoglu committed:
Added menubar filters for assignees and statuses

onuridrisoglu committed:
Priority column progress bar

onuridrisoglu committed:
Merge branch 'dev-attachment' into dev-smallchanges

Conflicts:
	bugrap/src/main/java/org/vaadin/bugrap/ui/ReportsModel.java
	bugrap/src/main/java/org/vaadin/bugrap/ui/ReportsView.java

onuridrisoglu committed:
Added | chars between buttons, and fixed the issue with distribution bar

marciopd commented on bugrap/pom.xml:
```diff
@@ -48,6 +48,11 @@
 	<dependencies>
+		<dependency>
+			<groupId>org.vaadin.addon</groupId>
+			<artifactId>report_distribution_bar</artifactId>
```
It's more common using '-' as a separator in the artifactId.

marciopd commented on bugrap/src/main/java/org/vaadin/bugrap/BaseModel.java:
```diff
@@ -4,19 +4,22 @@
+	private VaadinSession session;
 
-	public BaseModel(Navigator nav) {
+	public BaseModel(Navigator nav, VaadinSession vs) {
```
There's a way of getting the current Vaadin session, so you shouldn't need to be passing it around.

marciopd commented on bugrap/src/main/java/org/vaadin/bugrap/ui/ReportsModel.java:
```diff
@@ -108,6 +115,8 @@ public void setUploadedFilesToSave(List<Comment> uploadedFilesToSave) {
 		ReportsQuery query = new ReportsQuery();
 		query.project = project;
 		query.projectVersion = version;
+		query.reportAssignee = assigneeFilterMode == ASSIGNEE_ME ? getLoginUser() : null;
```
Avoid ternary operator.

onuridrisoglu replied:
I like ternary operator a lot :)

marciopd commented on bugrap/src/main/java/org/vaadin/bugrap/BaseModel.java:
```diff
@@ -28,7 +31,23 @@ protected Navigator getNavigator() {
 	}
+	
+	public Reporter getLoginUser() {
+		return (Reporter)getSessionVariable(SESSIONVAR_LOGINUSER);
```
Formatting.

marciopd commented on bugrap/src/main/java/org/vaadin/bugrap/ui/LoginView.java:
```diff
@@ -23,7 +22,7 @@ public LoginView(LoginModel m) {
 	@Override
 	public void enter(ViewChangeEvent event) {
-		if (BaseModel.loginUser != null)
+		if (model.getLoginUser() != null)
```
This if is looking strange. Is it correct?

onuridrisoglu replied:
it's actualy working fine, but I agree its looking kinda strange :)

marciopd commented on bugrap/src/main/java/org/vaadin/bugrap/ui/ReportsView.java:
```diff
@@ -47,24 +67,81 @@ public void enter(ViewChangeEvent event) {
+		menuStatus.getItems().get(MENUBARIDX_STATUS_ALL).setCommand(evt -> onStatusFilterSelected(true, true, Status.values()));
+		
+		MenuItem customStatusMenuItem = menuStatus.getItems().get(MENUBARIDX_STATUS_CUSTOM);
+		customStatusMenuItem.getChildren().get(MENUBARIDX_CUSTOMSTATUS_OPEN).setCommand(evt -> onStatusFilterSelected(false, evt.isChecked(), Status.OPEN));
```
getChildren() could be called only once.

marciopd commented on bugrap/src/main/java/org/vaadin/bugrap/ui/ReportsView.java:
```diff
@@ -47,24 +67,81 @@ public void enter(ViewChangeEvent event) {
+		menuStatus.getItems().get(MENUBARIDX_STATUS_ALL).setChecked(model.getStatusFilter().containsAll(Arrays.asList(Status.values())));
+		
+		MenuItem customStatusMenuItem = menuStatus.getItems().get(MENUBARIDX_STATUS_CUSTOM);
+		customStatusMenuItem.getChildren().get(MENUBARIDX_CUSTOMSTATUS_OPEN).setChecked(model.getStatusFilter().contains(Status.OPEN));
```
getChildren() could be called only once.

Consider other places where this comment could be applied.

marciopd commented on report_distribution_bar/src/test/java/org/vaadin/addon/report_distribution_bar/ReportDistributionBarTest.java:
```diff
@@ -0,0 +1,13 @@
+
+	@Test
+	public void thisAlwaysPasses() {
+		Assert.assertEquals(true, true);
```
Seems that are no real tests here. Should this file be removed?

marciopd commented on report_distribution_bar/src/main/java/org/vaadin/addon/report_distribution_bar/client/ReportDistributionBarWidget.java:
```diff
@@ -0,0 +1,47 @@
+	}
+
+	private String getWidth(long num, long denom) {
+		long width = (num * 90)/denom;
```
90 is a magical number here. It would be better to take it into a constant and give it a good name.

marciopd reviewed (changes requested):
Minor comments. Good work!

onuridrisoglu committed:
Pull Request Changes

The way session is read in Model classes is changed as well as some
format changes

marciopd reviewed (approved):
Good!
